src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -363,13 +364,60 @@ public void run(final RunNotifier notifier) {
             statement.evaluate();
         } catch (AssumptionViolatedException e) {
             testNotifier.addFailedAssumption(e);
-        } catch (StoppedByUserException e) {
-            throw e;
         } catch (Throwable e) {
-            testNotifier.addFailure(e);
```

Comment:
@NikitaMerinov So is there a problem with my suggestion to have `ParentRumner` notice that a stop was requested and throw a new `StoppedByUserException` (or,  better yet, do this in a listener after we add suite start/stop calls?